### PR TITLE
Implements the  Acceptance Strategy

### LINKED ISSUE #1: Acceptance Strategy needs implementation

### PR DESCRIPTION
**Pull request by tiagomsleao**
The function of the percentage is still linear instead of exponential
Closes #1

### CODE REVIEW

**Review by pmisteliac — DISMISSED** [review]
The strategy looks good, but I have some comments on the code formatting. Java variable and function names are in camel case by convention. So it would be reservationValue instead of reservation_value.
I don't mean to be overly criticizing but code quality accounts for 20% of our grade so I think we should pay attention to detail here because this is easy to achieve.

File: AitGroup25Assignment/src/ai2018/group25/Group25_AS.java [hunk 1]
```diff
@@ -34,21 +37,48 @@ public void init(NegotiationSession negotiationSession, OfferingStrategy offerin
...
 
 	
 	@Override
 	public Actions determineAcceptability() {
-		//TODO implement strategy here
+		// Accepting Strategy Implementation
+		
+		// Initialize variable
+		double myLastBidUtil = -1.0;
+		double right_limit = -1.0;
```
Comment (RensHam) [inline AitGroup25Assignment/src/ai2018/group25/Group25_AS.java]: As already said try to only use camelcase.

File: AitGroup25Assignment/src/ai2018/group25/Group25_AS.java [hunk 2]
```diff
@@ -34,21 +37,48 @@ public void init(NegotiationSession negotiationSession, OfferingStrategy offerin
...
+		
+		// Get my last bid and the bid I am planning on doing next
+		if (negotiationSession.getOwnBidHistory().getLastBidDetails()!=null) { // Cover the first case
```
Comment (RensHam) [inline AitGroup25Assignment/src/ai2018/group25/Group25_AS.java]: Documentation in code shouldn't be used to much. And please : 
```if (negotiationSession.getOwnBidHistory().getLastBidDetails() != null) {```

File: AitGroup25Assignment/src/ai2018/group25/Group25_AS.java [hunk 3]
```diff
@@ -34,21 +37,48 @@ public void init(NegotiationSession negotiationSession, OfferingStrategy offerin
...
+		
+		// The Right limit is the minimum between my last and my next bid
+		if (myLastBidUtil != -1.0) {
+			right_limit = Math.min(myLastBidUtil, myNextBidUtil);
+		} else {
+			right_limit = myNextBidUtil;
+		}
+		
+		// Get, according to time, the percentage of the interval not to accept
+		double percentage_interval_rejected = what_reject(negotiationSession.getTime()); // DO THIS FUNCTION
```
Comment (RensHam) [inline AitGroup25Assignment/src/ai2018/group25/Group25_AS.java]: Remove comment //DO THIS FUNCTION

File: AitGroup25Assignment/src/ai2018/group25/Group25_AS.java [hunk 4]
```diff
@@ -70,7 +102,16 @@ public String getName() {
 	
 	@Override
 	public String printParameters() {
-		return "[paramter 1: " + parameter1 + " ]";
+		return "[paramter 1: " + reservation_value + " ]";
 	}
-
+	
+	private double what_reject(double normalized_time) {
+		if ( normalized_time <= starts_falling ) {
```
Comment (RensHam) [inline AitGroup25Assignment/src/ai2018/group25/Group25_AS.java]: ```if (val == val2) {```

File: AitGroup25Assignment/src/ai2018/group25/Group25_AS.java [hunk 5]
```diff
@@ -70,7 +102,16 @@ public String getName() {
...
-		return "[paramter 1: " + parameter1 + " ]";
+		return "[paramter 1: " + reservation_value + " ]";
 	}
-
+	
+	private double what_reject(double normalized_time) {
+		if ( normalized_time <= starts_falling ) { 
+			return 1.0;
+		} else {
+			return ( -1/(1-starts_falling) )*normalized_time + ( 1/(1-starts_falling) ); // Still linear, transform into exponential
```
Comment (RensHam) [inline AitGroup25Assignment/src/ai2018/group25/Group25_AS.java]: It is more usefull to write javadoc above the function what it should do. Then this comment wouldn't be needed.

File: AitGroup25Assignment/src/ai2018/group25/Group25_AS.java [hunk 6]
```diff
@@ -34,21 +37,48 @@ public void init(NegotiationSession negotiationSession, OfferingStrategy offerin
...
+			reservation_value = RESERVATION_VALUE;
+			starts_falling = STARTS_FALLING;
 		}
 	}
 
 	
 	@Override
 	public Actions determineAcceptability() {
-		//TODO implement strategy here
+		// Accepting Strategy Implementation
```
Comment (RensHam) [inline AitGroup25Assignment/src/ai2018/group25/Group25_AS.java:51]: Remove useless comments.

**Review by RensHam — DISMISSED** [review]

File: AitGroup25Assignment/src/ai2018/group25/Group25_AS.java (hunk 4 above)
Comment (tiagomsleao) [reply AitGroup25Assignment/src/ai2018/group25/Group25_AS.java]: whats this?

File: AitGroup25Assignment/src/ai2018/group25/Group25_AS.java (hunk 6 above)
Comment (tiagomsleao) [reply AitGroup25Assignment/src/ai2018/group25/Group25_AS.java:51]: it is not useless...

File: AitGroup25Assignment/src/ai2018/group25/ConstBid.java [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+	/**
+	 * k in [0, 1]. For k = 0 the agent starts with a bid of maximum utility
+	 */
+	private double k;
+	/** Maximum target utility */
+	private double Pmax;
+	/** Minimum target utility */
+	private double Pmin;
+	/** Concession factor */
+	private double e;
```
Comment (RensHam) [inline AitGroup25Assignment/src/ai2018/group25/ConstBid.java:45]: I would go for
`private double concessionFactor` but that is upto you

File: AitGroup25Assignment/src/ai2018/group25/ConstBid.java [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+		}*/
+		nextBid = omStrategy.getBid(outcomespace, 0.7);
+		return nextBid;
+	}
+
+	/**
+	 * From [1]:
+	 * 
+	 * A wide range of time dependent functions can be defined by varying the
+	 * way in which f(t) is computed. However, functions must ensure that 0 <=
```
Comment (RensHam) [inline AitGroup25Assignment/src/ai2018/group25/ConstBid.java:124]: :)

File: AitGroup25Assignment/src/ai2018/group25/ConstBid.java (hunk 7 above)
Comment (tiagomsleao) [reply AitGroup25Assignment/src/ai2018/group25/ConstBid.java:45]: I justed used what was done and commented it all as you should have seen...

**Pull request merged**